src/main.rs

```diff
@@ -215,9 +215,15 @@ fn config_options<'a, 'b>() -> App<'a, 'b> {
                         .help("the secret to sign the JWT with. Can be prefixed with @ to read from a binary file")
                         .takes_value(true)
                         .long("secret")
-                        .short("S")
-                        .required(true),
-                ),
+                        .short("S"),
+                ).arg(
+                    Arg::with_name("secret-b64")
```

Comment:
If I may suggest, a more "standard" approach might be to add an arg called `encoding` which accepts standard names for supported encodings, such as `base64`.

e.g. `--encoding=base64` / `-e base64`

Reply:
Would you suggest that -e allow any of {file,string,base64}? Or just -e base64, and continue as-is with the "if it starts with @, it's a file"?

(I'm not opposed to doing this, I've just not seen this approach before.)

Reply:
Good question. My thought was just for it to be used for encodings (base64 to start, but maybe someone could add others in the future, i.e. base32?). I'm not sure how string/file should be handled. My intuition is they shouldn't be included in the `-e` arg because they're not encodings (unless you want to consider ASCII/Unicode encodings). 

Maybe @mike-engel has some input.

Reply:
Yeah, it would be better, I think, to be more generic in this case, since there will be more encodings as time goes on.

However, `--base64` is *too* generic, in that it could apply to many different options. My preference, I think, would be `--secret-encoding=value`, where value would be an enum of possible options, like `algorithm` is (`jwt-cli` only supports some, but not all algorithms!)

So, that would probably be

```rust
enum SecretEncoding {
  Binary,
  Base64,
}
```

Thoughts @therealklanni and @ismith?

Reply:
I like that approach

Reply:
SGTM! Sorry for the delay, was busy conferencing.